docs/meeting-minutes/TrenchBoot-AEM-January24-2023.md

```diff
@@ -0,0 +1,62 @@
+## Attendees
+
+- Daniel P. Smith (Apertus Solutions)
+- Andrew Cooper (Citrix)
+- Krystian Hebel (3mdeb)
+- Michał Żygowski (3mdeb)
+
+## Agenda
+
+- Review the current state of TrenchBoot boot protocol being upstreamed to GRUB
+  and Linux kernel
+- Discuss the scope of the work needed to align Qubes OS Anti Evil Maid
+  solution based on TrenchBoot with the most recent boot protocol version
+- Discuss possibilities to implement DRTM launch on S3 resume in Xen
+
+## Meeting Minutes
+
+Current approach of booting Linux with TrenchBoot:
+
+- meet Linux demand so that anything can do EBS and DRTM can still work
+- Linux doing EBS to gather necessary info from firmware
+- bootloader to put slaunch code to memory
+- the code being put to memory is going to be standardized and does everything
+  what DRTM technology needs
+- standardize an input table to be used for generic stub not specific to any
+  bootloader/kernel
+
+Current Xen status:
+
+- Xen efi needs fixing in order to work without Xen's efi-stub, so it would be
+  better to follow Linux's approach of jumping into Xen and Xen calling out to
+  slstub
```

Comment:
Currently, Xen's EFI code (like linux) depends on being able to use boot services, and expects to call EBS itself.

Previously, we had a plan to try and make Xen able to boot without access to boot services.  And while this is still a good idea, the point I was trying to make is that because we must support boot services for Linux efi-stub, using the same approach in Xen would be a less invasive change overall, and therefore easier to upstream.

i.e. "making Xen work without boot services" can (and should) now be orthogonal to "making Xen work with trenchboot".

Reply:
Yes, initially we wanted to get rid of UEFI completely in Xen, but given the new approach of booting Linux with TrenchBoot, it doesn't make much sense to pursue deUEFIcation anymore. This is what I understood too. Do you request any change in the file @andyhhp ?

Reply:
I'm suggesting another approach:
- Xen is built as usual, with changes we made in the first phase, but this time we use `xen.efi`
- GRUB, when started with slaunch activated, doesn't care about "EFI boot services" tag (conditional added before https://github.com/TrenchBoot/grub/blob/master/grub-core/loader/multiboot_mbi2.c#L257)
  - BS are terminated (https://github.com/TrenchBoot/grub/blob/master/grub-core/loader/multiboot_mbi2.c#L943)
  - UEFI MB2 entry point is **not** taken (https://github.com/TrenchBoot/grub/blob/master/grub-core/loader/multiboot_mbi2.c#L362)
  - memory info is passed through Memory map tag (https://github.com/TrenchBoot/grub/blob/master/grub-core/loader/multiboot_mbi2.c#L810) and Basic memory info tag (https://github.com/TrenchBoot/grub/blob/master/grub-core/loader/multiboot_mbi2.c#L832)
- Xen boots through legacy MB2 path (actually, it starts with TXT entry that quickly jumps into MB2 path) with all necessary info provided by GRUB

That way there would be no difference in Xen code between UEFI and legacy boot at all.

Reply:
"Xen efi needs fixing in order to work without Xen's efi-stub," doesn't really parse IMO.   We don't really have an efi-stub like Linux does.  We've just got a bit of code which is compiled to MSABI.

Reply:
@andyhhp so, could you please suggest how this sentence should look like?

Also what do you think about @krystian-hebel approach?